commands/service_execute.go

```diff
@@ -36,33 +33,36 @@ func newServiceExecuteCmd(e ServiceExecutor) *serviceExecuteCmd {
 		RunE:    c.runE,
 	})
 	c.cmd.Flags().StringVarP(&c.taskKey, "task", "t", c.taskKey, "Run the given task")
-	c.cmd.Flags().VarP(xpflag.NewStringToStringValue(&c.executeData, nil), "data", "d", "data required to run the task")
-	c.cmd.Flags().StringVarP(&c.jsonFile, "json", "j", c.jsonFile, "Path to a JSON file containing the data required to run the task")
+	c.cmd.Flags().StringVarP(&c.jsonFile, "json", "j", c.jsonData, "JSON containing the data required to run the task")
+	c.cmd.Flags().StringVarP(&c.jsonFile, "json-file", "f", c.jsonFile, "Path to a JSON file containing the data required to run the task")
+
+	c.cmd.MarkPersistentFlagRequired("task")
 	return c
 }
 
 func (c *serviceExecuteCmd) preRunE(cmd *cobra.Command, args []string) error {
-	if cmd.Flag("data").Changed && cmd.Flag("json").Changed {
-		return errors.New("Only one of '--data' or '--json' options can be specified")
+	if cmd.Flag("json").Changed && cmd.Flag("json-file").Changed {
```

Comment:
What's the reason behind this modification ? the `--data` is really good so why removing it in favor of a raw json (which is really unfriendly). I would remove this change

Reply:
The problem with --data is casting method. We need to know service definition in order to cast data value to proper types. With service definition spread over the network, we need to download service definition first in order to cast the data.
It is complicated right now for writing PoC and moreover for feature usage. I spoke with @NicolasMahe about it and we agree to change data to JSON.